conformance/lang/expressions/literals/int_literal.balt

```diff
@@ -0,0 +1,502 @@
+
+    x = 1234567890F;
+    io:println(x is int); // @output false
+    io:println(x is float); // @output true
+    io:println(x is decimal); // @output false
+
+    x = 786.218f;
+    io:println(x is int); // @output false
+    io:println(x is float); // @output true
+    io:println(x is decimal); // @output false
+}
+
+Test-Case: error
+Description: Test the basic type when the numeric-literal includes a FloatTypeSuffix.
+Labels: decimal, FloatTypeSuffix, float, int
+
+public function main() {
+    int x = 0f; // @error expected an 'int' but found a 'float'
+
+    decimal y = 12345678.90F; // @error expected a 'decimal' but found a 'float'
+
+    int z = 786f; // @error expected an 'int' but found a 'float'
+}
+
+Test-Case: output
+Description: Test the basic type when the numeric-literal includes a DecimalTypeSuffix.
+Labels: any, DecimalTypeSuffix, decimal, float, int, is-expr
+
+public function main() {
+    any x = 0d;
+    io:println(x is int); // @output false
+    io:println(x is float); // @output false
+    io:println(x is decimal); // @output true
+
+    x = 1234567890D;
+    io:println(x is int); // @output false
+    io:println(x is float); // @output false
+    io:println(x is decimal); // @output true
+
+    x = 786.218d;
+    io:println(x is int); // @output false
+    io:println(x is float); // @output false
+    io:println(x is decimal); // @output true
+}
+
+Test-Case: error
+Description: Test the basic type when the numeric-literal includes a DecimalTypeSuffix.
+Labels: DecimalTypeSuffix, decimal, float, int
+
+public function main() {
+    int x = 0d; // @error expected an 'int' but found a 'decimal'
+
+    float y = 12345678.90D; // @error expected an 'float' but found a 'decimal'
+
+    int z = 786d; // @error expected an 'int' but found a 'decimal'
+}
+
+Test-Case: output
+Description: Test the basic type when the numeric-literal is a HexFloatingPointLiteral.
+Labels: any, decimal, float, HexFloatingPointLiteral, int, is-expr
+
+public function main() {
+    any x = 0xabc12.def;
+    io:println(x is int); // @output false
+    io:println(x is float); // @output true
+    io:println(x is decimal); // @output false
+
+    x = 0x.2131ef;
+    io:println(x is int); // @output false
+    io:println(x is float); // @output true
+    io:println(x is decimal); // @output false
+
+    x = 0x123.456cP4;
+    io:println(x is int); // @output false
+    io:println(x is float); // @output true
+    io:println(x is decimal); // @output false
+}
+
+Test-Case: error
+Description: Test the basic type when the numeric-literal is a HexFloatingPointLiteral.
+Labels: decimal, float, HexFloatingPointLiteral, int
+
+public function main() {
+    decimal x = 0xabc12.def; // @error expected an 'decimal' but found a 'float'
+
+    decimal y = 0x.2131ef; // @error expected an 'decimal' but found a 'float'
+
+    int z = 0x123.456cP4; // @error expected an 'int' but found a 'float'
+}
+
+Test-Case: output
```

Comment:
Shall we move this to after float? Shall we also include a separate test for hex int literals?